app/assets/src/components/views/amr_heatmap/AMRHeatmapControls.jsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Divider } from "~/components/layout";
+import { Dropdown } from "~ui/controls/dropdowns";
+
+import cs from "./amr_heatmap_view.scss";
+
+export default class AMRHeatmapControls extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  generateChangeFunction(filter) {
+    const { selectedOptions, onSelectedOptionsChange } = this.props;
+    const changeFunction = option => {
+      if (option !== selectedOptions[filter]) {
+        onSelectedOptionsChange({ [filter]: option });
+      }
+    };
+    return changeFunction;
+  }
+
+  renderFilterDropdowns() {
+    const { filters, selectedOptions, hasData } = this.props;
+    const filtersList = [];
+    filters.forEach((filterOptionData, filter) => {
+      filtersList.push(
+        <div className="col s3" key={filter}>
+          <Dropdown
+            fluid
+            rounded
+            options={filterOptionData.options}
+            onChange={this.generateChangeFunction(filter)}
```

Comment:
What you have works. You can pass `filter` into the function like this, which is a bit more concise:

```

onChange={option => this.handleFilterChange(option, filter)}

// Showing the handleFilterChange function for clarity
const handleFilterChange = (option, filter) => {
  if (option !== selectedOptions[filter]) {
    onSelectedOptionsChange({ [filter]: option });
  }
};
```